src/Domain/PostalAddress.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace MatchBot\Domain;
+
+use Doctrine\ORM\Mapping\Column;
+use Doctrine\ORM\Mapping\Embeddable;
+use MatchBot\Application\Assertion;
+
+/**
+ * Value object for convenience and organisation but with very minimal validation since data is
+ * currently all entered via Salesforce so we can't throw back to the UI for any bad data.
+ */
+#[Embeddable]
+readonly class PostalAddress
+{
+    public static function of(
+        string $line1,
+        ?string $line2,
+        ?string $city,
+        ?string $postalCode,
+        ?string $country,
+    ): self {
+        return new self($line1, $line2, $city, $postalCode, $country);
+    }
+
+    private function __construct(
+        #[Column(nullable: true)]
+        public string $line1,
```

Comment:
looks like Doctrine recognises that this property is non-nullable and gives the Charity a null instead of an address object if line1 is null in the db, which is what I want.